TTHAnalysis/python/treeReAnalyzer.py

```diff
@@ -70,6 +70,8 @@ def __init__(self,event,prefix,index=None):
         self._event = event
         self._prefix = prefix+"_"
         self._index = index
+        self.p4vec = ROOT.TLorentzVector()
+        self.p4vec.SetPtEtaPhiM(self.pt,self.eta,self.phi,self.mass)

```

Comment:
I am not sure this is safe, for sure it's not necessary. I agree having self.p4vec persistent in the class, but we can create it (or update it if it already exists) when p4() is called, can't we?

def p4(self):
    if not hasattr(self,'p4vec'): self.p4vec = ROOT.TLorentzVector()
    p4vec.SetPtEtaPhiM(self.pt,self.eta,self.phi,self.mass)
    return self.p4vec

and then delete it in the destructor only if it's there.

@gpetruc, is this fine? Are we sure that this memory address will not be overwritten by other calls, like other objects in the same collection?